### ISSUE TITLE
Rultor Dies when Acting on a deleted Comment

### ISSUE THREAD

**Issue by original-brownbear** [yegor256/rultor#1125]
This Sequence of events appears to kill Rultor as it is currently down with message:

```
java.lang.AssertionError: HTTP response status is not equal to 200:
404 Not Found [https://api.github.com/repos/CheckHealthCo/dx2/issues/comments/246167328]
Server: GitHub.com
Date: Sun, 11 Sep 2016 10:20:58 GMT
Content-Type: application/json; charset=utf-8
Transfer-Encoding: chunked
Status: 404 Not Found
X-RateLimit-Limit: 5000
X-RateLimit-Remaining: 4845
X-RateLimit-Reset: 1473591681
X-OAuth-Scopes: admin:org, admin:public_key, admin:repo_hook, delete_repo, gist, notifications, repo, user
X-Accepted-OAuth-Scopes: repo
X-GitHub-Media-Type: github.v3
Access-Control-Expose-Headers: ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval
Access-Control-Allow-Origin: *
Content-Security-Policy: default-src 'none'
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 369E0910:117F:83213B8:57D5300A

{"message":"Not Found","documentation_url":"https://developer.github.com/v3"}
Expected: HTTP response with status 200
     but: was <404 Not Found [https://api.github.com/repos/CheckHealthCo/dx2/issues/comments/246167328]
Server: GitHub.com
Date: Sun, 11 Sep 2016 10:20:58 GMT
Content-Type: application/json; charset=utf-8
Transfer-Encoding: chunked
Status: 404 Not Found
X-RateLimit-Limit: 5000
X-RateLimit-Remaining: 4845
X-RateLimit-Reset: 1473591681
X-OAuth-Scopes: admin:org, admin:public_key, admin:repo_hook, delete_repo, gist, notifications, repo, user
X-Accepted-OAuth-Scopes: repo
X-GitHub-Media-Type: github.v3
Access-Control-Expose-Headers: ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval
Access-Control-Allow-Origin: *
Content-Security-Policy: default-src 'none'
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 369E0910:117F:83213B8:57D5300A

{"message":"Not Found","documentation_url":"https://developer.github.com/v3"}>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at com.jcabi.http.response.RestResponse.assertStatus(RestResponse.java:111)
	at com.jcabi.github.RtJson.fetch_aroundBody0(RtJson.java:77)
	at com.jcabi.github.RtJson$AjcClosure1.run(RtJson.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at com.jcabi.aspects.aj.MethodLogger.wrap(MethodLogger.java:207)
	at com.jcabi.aspects.aj.MethodLogger.ajc$inlineAccessMethod$com_jcabi_aspects_aj_MethodLogger$com_jcabi_aspects_aj_MethodLogger$wrap(MethodLogger.java:1)
	at com.jcabi.aspects.aj.MethodLogger.wrapClass(MethodLogger.java:136)
	at com.jcabi.github.RtJson.fetch(RtJson.java:77)
	at com.jcabi.github.RtComment.json_aroundBody6(RtComment.java:116)
	at com.jcabi.github.RtComment$AjcClosure7.run(RtComment.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at com.jcabi.aspects.aj.MethodLogger.wrap(MethodLogger.java:207)
	at com.jcabi.aspects.aj.MethodLogger.ajc$inlineAccessMethod$com_jcabi_aspects_aj_MethodLogger$com_jcabi_aspects_aj_MethodLogger$wrap(MethodLogger.java:1)
	at com.jcabi.aspects.aj.MethodLogger.wrapClass(MethodLogger.java:136)
	at com.jcabi.github.RtComment.json(RtComment.java:116)
	at com.jcabi.github.SmartJson.json_aroundBody4(SmartJson.java:104)
	at com.jcabi.github.SmartJson$AjcClosure5.run(SmartJson.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at com.jcabi.aspects.aj.MethodLogger.wrap(MethodLogger.java:207)
	at com.jcabi.aspects.aj.MethodLogger.ajc$inlineAccessMethod$com_jcabi_aspects_aj_MethodLogger$com_jcabi_aspects_aj_MethodLogger$wrap(MethodLogger.java:1)
	at com.jcabi.aspects.aj.MethodLogger.wrapClass(MethodLogger.java:136)
	at com.jcabi.github.SmartJson.json(SmartJson.java:104)
	at com.jcabi.github.SmartJson.value_aroundBody6(SmartJson.java:120)
	at com.jcabi.github.SmartJson$AjcClosure7.run(SmartJson.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at com.jcabi.aspects.aj.MethodLogger.wrap(MethodLogger.java:207)
	at com.jcabi.aspects.aj.MethodLogger.ajc$inlineAccessMethod$com_jcabi_aspects_aj_MethodLogger$com_jcabi_aspects_aj_MethodLogger$wrap(MethodLogger.java:1)
	at com.jcabi.aspects.aj.MethodLogger.wrapClass(MethodLogger.java:136)
	at com.jcabi.github.SmartJson.value(SmartJson.java:120)
	at com.jcabi.github.SmartJson.text_aroundBody0(SmartJson.java:81)
	at com.jcabi.github.SmartJson$AjcClosure1.run(SmartJson.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at com.jcabi.aspects.aj.MethodLogger.wrap(MethodLogger.java:207)
	at com.jcabi.aspects.aj.MethodLogger.ajc$inlineAccessMethod$com_jcabi_aspects_aj_MethodLogger$com_jcabi_aspects_aj_MethodLogger$wrap(MethodLogger.java:1)
	at com.jcabi.aspects.aj.MethodLogger.wrapClass(MethodLogger.java:136)
	at com.jcabi.github.SmartJson.text(SmartJson.java:81)
	at com.jcabi.github.Comment$Smart.body_aroundBody2(Comment.java:137)
	at com.jcabi.github.Comment$Smart$AjcClosure3.run(Comment.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at com.jcabi.aspects.aj.MethodLogger.wrap(MethodLogger.java:207)
	at com.jcabi.aspects.aj.MethodLogger.ajc$inlineAccessMethod$com_jcabi_aspects_aj_MethodLogger$com_jcabi_aspects_aj_MethodLogger$wrap(MethodLogger.java:1)
	at com.jcabi.aspects.aj.MethodLogger.wrapClass(MethodLogger.java:136)
	at com.jcabi.github.Comment$Smart.body(Comment.java:137)
	at com.rultor.agents.github.Reports.process(Reports.java:101)
	at com.rultor.agents.AbstractAgent.execute(AbstractAgent.java:78)
	at com.rultor.spi.Agent$Iterative.execute(Agent.java:84)
	at com.rultor.Routine.process(Routine.java:165)
	at com.rultor.Routine.safe_aroundBody0(Routine.java:145)
	at com.rultor.Routine$AjcClosure1.run(Routine.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at com.jcabi.aspects.aj.MethodInterrupter.wrap(MethodInterrupter.java:115)
	at com.rultor.Routine.safe(Routine.java:140)
	at com.rultor.Routine.run(Routine.java:113)
	at com.jcabi.log.VerboseRunnable.run(VerboseRunnable.java:198)
	at com.jcabi.aspects.aj.MethodScheduler$Service$1.run(MethodScheduler.java:179)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:304)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:178)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at com.jcabi.log.VerboseThreads$Wrap.run(VerboseThreads.java:220)
	at java.lang.Thread.run(Thread.java:745)
```

Caused by:

1. Comment
2. Rultor receives notification
3. User Deletes Comment
4. Rultor tries to act on the deleted comment, as it was deleted after Rultor received the notification about it
5. Above error

**Comment by original-brownbear** [comment]
@alex-palevsky this is a bug.

**Comment by alex-palevsky** [comment]
> @alex-palevsky this is a bug.

@original-brownbear I just added bug tag here

**Comment by original-brownbear** [comment]
@yegor256 #1126 should get us in the clear here for now.

**Comment by alex-palevsky** [comment]
@original-brownbear attached this issue to milestone "2.0" (let me know if this is wrong)

**Comment by alex-palevsky** [comment]
@original-brownbear many thanks, **30 mins** added to your acc for reporting this bug, pmt ID `AP-9GE99756MD829362F`

**Comment by yegor256** [comment]
@original-brownbear I guess it's fixed

**Comment by original-brownbear** [comment]
@yegor256 jup :)

**Comment by alex-palevsky** [comment]
@original-brownbear once `1125-c2e86b18`/#1127 puzzle is resolved (later, in another ticket), this ticket will be fully complete

**Comment by alex-palevsky** [comment]
@original-brownbear last puzzled solved `1125-c2e86b18`/#1127

**Comment by 0pdd** [comment]
@original-brownbear the puzzle [#1370](https://github.com/yegor256/rultor/issues/1370) is still not solved.
